specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -788,6 +940,165 @@
+            "$ref": "./examples/NSPCheckMembersPost.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Network/locations/{location}/queryNetworkSecurityPerimeter": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/LocationParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "QueryNetworkSecurityPerimeters"
+        ],
+        "operationId": "NetworkSecurityPerimeters_Query",
+        "description": "Action to query network security parameter",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QueryNSPReqRes"
+            },
+            "description": "Parameter supplied to query network security parameter."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Query NSP using parameter guid, no resource created or updated.",
+            "schema": {
+              "$ref": "#/definitions/QueryNSPReqRes"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Post Query NSP": {
+            "$ref": "./examples/QueryNSP.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkSecurityPerimeters/{networkSecurityPerimeterName}/profiles/{profileName}/accessRules/{accessRuleName}/reconcile": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterNameParameter"
+        },
+        {
+          "$ref": "#/parameters/ProfileNameParameter"
+        },
+        {
+          "$ref": "#/parameters/AccessRuleNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "NspAccessRulesReconcile"
+        ],
+        "operationId": "NspAccessRulesReconcile_Post",
+        "description": "Reconcile NSP access rules",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NspAccessRuleReconcile"
+            },
+            "description": "Parameters for NSP access rule reconcile"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Reconcile NSP access rule.",
+            "schema": {
+              "$ref": "#/definitions/NspAccessRuleReconcile"
```

Comment:
>NspAccessRuleReconcile [](http://example.com/codeflow?start=37&length=22)

same here again

Reply:
The response contains similar data with a few more added keys so we are reusing the object.

Spec: https://msazure.visualstudio.com/One/_git/Networking-MNC?path=/doc/Design-Docs/Network-Security-Perimeter-Design-Specification.md&version=GBmaster&_a=preview&anchor=profile-reconciliation